packages/netsuite-adapter/src/filters/element_references.ts

```diff
@@ -127,16 +127,23 @@ const generateServiceIdToElemID = async (
 const resolveRelativePath = (absolutePath: string, relativePath: string): string =>
   osPath.resolve(osPath.dirname(absolutePath), relativePath)
 
+const getFileNameFromPath = (path: string):string =>
+  path.split(FILE_CABINET_PATH_SEPARATOR)
+    .slice(-1)[0]
+    .split('.')[0]
+
 const getServiceElemIDsFromPaths = (
   foundReferences: string[],
   serviceIdToElemID: ServiceIdRecords,
   customRecordFieldsToServiceIds: ServiceIdRecords,
   element: InstanceElement,
+  fileNames: Set<string>,
 ): ElemID[] =>
   foundReferences
     .flatMap(ref => {
-      if (pathPrefixRegex.test(ref)) {
-        const absolutePath = resolveRelativePath(element.value[PATH], ref)
+      if (fileNames.has(getFileNameFromPath(ref))) {
+        const prefixedRef = pathPrefixRegex.test(ref) ? ref : FILE_CABINET_PATH_SEPARATOR.concat(ref)
+        const absolutePath = resolveRelativePath(element.value[PATH], prefixedRef)
```

Comment:
we don't need to resolve the relative path in the second case
```suggestion
        const absolutePath = pathPrefixRegex.test(ref) ? resolveRelativePath(element.value[PATH], ref) : FILE_CABINET_PATH_SEPARATOR.concat(ref)
```